File: helloPerson.cpp
#include <iostream> // Input/Output stream functionality
#include <string> // C++ built-in character array container

class person{
    // First we define our data members, which by default can only be modified by our
    // member functions that we define below.
    std::string title, firstName, lastName;
    int age;
    //Anything below 'public' will be accessible to us in other parts of the code.
    public:
        // Just as in python, we define a constructor that initializes the object.
        // In C++ this function is just the name of the class
        person(std::string title, std::string firstName, std::string lastName, int age) {
            this->title = title;
            this->firstName = firstName;
            this->lastName = lastName;
            this->age = age;
        }
        // Below we will define the same four member functions we had in our python code.
        // in C++ we have to specify the return type of each function. If we don't want the
        // function to return anything, we declare it a void type.
        void greetInformal () {
            std::cout << "Hello, " << this->firstName << "!\n";
        }

        void greetFormal () {
            std::cout << "Hello, " << this->title << " " << this->lastName << "!\n";
        }

        void howOld() {
            std::cout << this->firstName << " is " << this->age << " years old.\n";
        }
        void birthday() {
            this->age +=1;
            std::cout << "Happy Birthday, " << this->firstName << "! You are now "
                      << this->age << " years old.\n";
        }
};

// This is where our computer will start executing code
int main() {
    // Storage for our user input
    std::string title, firstName, lastName;
    int age;

    // Get user input
    std::cout << "What is your title?: ";
    std::cin >> title;
    std::cout << "What is your first name?: ";
    std::cin >> firstName;
    std::cout << "What is your last name?: ";
    std::cin >> lastName;
    std::cout << "How old are you?: ";
    std::cin >> age;

    // Create the class object
    person you (title, firstName, lastName, age);

    // Execute the member functions
    you.greetInformal();
    you.greetFormal();
    you.howOld();
    you.birthday();
    you.howOld();

    // End the program

    return 0;
}

File: helloPerson.py
# -*- coding: utf-8 -*-
"""
Created on Wed Feb 12 11:09:44 2020

@author: skyle
"""

# This is a class declaration in Python. It will allow us to create objects called person
class person:
    # This is the initializer function. It is called automatically when a person object is created.
    # The values after the equals signs are default values that will fill in data not provided.
    #Every member function must have self as the first thing in the list of function parameters.
    def __init__(self, title = "Mr.", firstName = "John", lastName = "Doe", age = 42):
        self.title = title
        self.firstName = firstName
        self.lastName = lastName
        self.age = age
        
    # This method will great the person by their first name
    def greetInformal(self):
        print("Hello,", self.firstName, "\b!")
        
    # This method will greet the person by their title and last name
    def greetFormal(self):
        print("Hello,", self.title, self.lastName, "\b!")
        
    # This method will tell you the person's age
    def howOld(self):
        print(self.firstName, "is", self.age, "years old!")
        
    # This method will tell the person happy birthday and increase their age by 1
    def birthday(self):
         self.age = self.age + 1
         print("Happy Birthday,", self.firstName, "\b! You are now", self.age, "years old!")
         
# Get some input from the user about who they are.
title = input("What is your title?: ")
firstName = input("What is your first name?: ")
lastName = input("What is your last name?: ")
age = input("How old are you?: ")

# Create the class object
you = person(title, firstName, lastName, int(age))

#Call the class member functions on the object we just created
you.greetInformal()
you.greetFormal()
you.howOld()
you.birthday()
you.howOld()

# Test what happens if we try to change data members directly
you.age = you.age -1
you.howOld()    